### ISSUE TITLE
Thumbnail generate error

### ISSUE THREAD

**Issue by kevinlwhlwh** [ryuzee/SlideHub#378]
Two issue occurred when uploading slide in PDF

```bash
I, [2018-07-09T21:47:09.364023 #1029]  INFO -- : ["/tmp/d20180709-1029-1v9nobj/2c4023fd76125cb1d459e20f25f6398f.pdf", "/tmp/d20180709-1029-1v9nobj/slide-01.jpg", "/tmp/d20180709-1029-1v9nobj/slide-01.ppm", "/tmp/d20180709-1029-1v9nobj/slide-02.jpg", "/tmp/d20180709-1029-1v9nobj/slide-02.ppm", "/tmp/d20180709-1029-1v9nobj/slide-03.jpg", "/tmp/d20180709-1029-1v9nobj/slide-03.ppm", "/tmp/d20180709-1029-1v9nobj/slide-04.jpg", "/tmp/d20180709-1029-1v9nobj/slide-04.ppm", "/tmp/d20180709-1029-1v9nobj/slide-05.jpg", "/tmp/d20180709-1029-1v9nobj/slide-05.ppm", "/tmp/d20180709-1029-1v9nobj/slide-06.jpg", "/tmp/d20180709-1029-1v9nobj/slide-06.ppm", "/tmp/d20180709-1029-1v9nobj/slide-07.jpg", "/tmp/d20180709-1029-1v9nobj/slide-07.ppm", "/tmp/d20180709-1029-1v9nobj/slide-08.jpg", "/tmp/d20180709-1029-1v9nobj/slide-08.ppm", "/tmp/d20180709-1029-1v9nobj/slide-09.jpg", "/tmp/d20180709-1029-1v9nobj/slide-09.ppm", "/tmp/d20180709-1029-1v9nobj/slide-10.jpg", "/tmp/d20180709-1029-1v9nobj/slide-10.ppm", "/tmp/d20180709-1029-1v9nobj/slide-11.jpg", "/tmp/d20180709-1029-1v9nobj/slide-11.ppm", "/tmp/d20180709-1029-1v9nobj/slide-12.jpg", "/tmp/d20180709-1029-1v9nobj/slide-12.ppm", "/tmp/d20180709-1029-1v9nobj/slide-13.jpg", "/tmp/d20180709-1029-1v9nobj/slide-13.ppm", "/tmp/d20180709-1029-1v9nobj/slide-14.jpg", "/tmp/d20180709-1029-1v9nobj/slide-14.ppm", "/tmp/d20180709-1029-1v9nobj/slide-15.jpg", "/tmp/d20180709-1029-1v9nobj/slide-15.ppm", "/tmp/d20180709-1029-1v9nobj/slide-16.jpg", "/tmp/d20180709-1029-1v9nobj/slide-16.ppm", "/tmp/d20180709-1029-1v9nobj/slide-17.jpg", "/tmp/d20180709-1029-1v9nobj/slide-17.ppm"]
I, [2018-07-09T21:47:09.368901 #1029]  INFO -- : Generating thumbnails
I, [2018-07-09T21:47:09.368992 #1029]  INFO -- : ["/tmp/d20180709-1029-1v9nobj/slide-01.jpg", "/tmp/d20180709-1029-1v9nobj/slide-02.jpg", "/tmp/d20180709-1029-1v9nobj/slide-03.jpg", "/tmp/d20180709-1029-1v9nobj/slide-04.jpg", "/tmp/d20180709-1029-1v9nobj/slide-05.jpg", "/tmp/d20180709-1029-1v9nobj/slide-06.jpg", "/tmp/d20180709-1029-1v9nobj/slide-07.jpg", "/tmp/d20180709-1029-1v9nobj/slide-08.jpg", "/tmp/d20180709-1029-1v9nobj/slide-09.jpg", "/tmp/d20180709-1029-1v9nobj/slide-10.jpg", "/tmp/d20180709-1029-1v9nobj/slide-11.jpg", "/tmp/d20180709-1029-1v9nobj/slide-12.jpg", "/tmp/d20180709-1029-1v9nobj/slide-13.jpg", "/tmp/d20180709-1029-1v9nobj/slide-14.jpg", "/tmp/d20180709-1029-1v9nobj/slide-15.jpg", "/tmp/d20180709-1029-1v9nobj/slide-16.jpg", "/tmp/d20180709-1029-1v9nobj/slide-17.jpg"]
/opt/application/current/lib/slide_hub/convert_util.rb:57:in `read': No such file or directory @ rb_sysopen - /tmp/d20180709-1029-1v9nobj/1.txt (Errno::ENOENT)
	from /opt/application/current/lib/slide_hub/convert_util.rb:57:in `block in pdf_to_transcript'
	from /opt/application/current/lib/slide_hub/convert_util.rb:52:in `times'
	from /opt/application/current/lib/slide_hub/convert_util.rb:52:in `pdf_to_transcript'
	from /opt/application/current/lib/slide_hub/convert_procedure.rb:86:in `convert_to_transcript'
	from /opt/application/current/lib/slide_hub/convert_procedure.rb:27:in `block in run'
	from /root/.rbenv/versions/2.5.0/lib/ruby/2.5.0/tmpdir.rb:89:in `mktmpdir'
	from /opt/application/current/lib/slide_hub/convert_procedure.rb:19:in `run'
	from /opt/application/current/lib/slide_hub/batch.rb:19:in `block in execute'
	from /opt/application/current/lib/slide_hub/batch.rb:16:in `each'
	from /opt/application/current/lib/slide_hub/batch.rb:16:in `execute'
	from /opt/application/current/vendor/bundle/ruby/2.5.0/gems/railties-5.2.0/lib/rails/commands/runner/runner_command.rb:41:in `<main>'
	from /opt/application/current/vendor/bundle/ruby/2.5.0/gems/railties-5.2.0/lib/rails/commands/runner/runner_command.rb:41:in `eval'
	from /opt/application/current/vendor/bundle/ruby/2.5.0/gems/railties-5.2.0/lib/rails/commands/runner/runner_command.rb:41:in `perform'
	from /opt/application/current/vendor/bundle/ruby/2.5.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /opt/application/current/vendor/bundle/ruby/2.5.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /opt/application/current/vendor/bundle/ruby/2.5.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /opt/application/current/vendor/bundle/ruby/2.5.0/gems/railties-5.2.0/lib/rails/command/base.rb:65:in `perform'
	from /opt/application/current/vendor/bundle/ruby/2.5.0/gems/railties-5.2.0/lib/rails/command.rb:46:in `invoke'
	from /opt/application/current/vendor/bundle/ruby/2.5.0/gems/railties-5.2.0/lib/rails/commands.rb:18:in `<top (required)>'
	from bin/rails:4:in `require'
	from bin/rails:4:in `<main>'
```

Also see warning with 

```bash
stdin: is not a tty
```

tried to fix but cannot help.

Basic Info:
Version: `V1.6.2`
Cloud: `AWS S3`
DB : `MySQL`
SQS: `AWS SQS`

**Comment by ryuzee** [comment]
@kevinlwhlwh 

Thanks for cutting the ticket and sorry for the bother.
The failure is caused by pdftotext command in lib/slide_hub/convert_util L54. From error messages, pdftotext did not extract texts from the pdf. 

Did you encounter the problem with any slides or a specific slide?
If the problem occurred with specific slide, could you please provide me a copy of that slide?

**Comment by ryuzee** [comment]
@kevinlwhlwh 

I've added tiny workaround to the app and deployed the new container as 1.6.3.
Changes can be seen at https://github.com/ryuzee/SlideHub/pull/379

**Comment by kevinlwhlwh** [comment]
@ryuzee thanks for your instant update. Sorry the pdf include some confidential data that i can not send it to you, however i can describe what inside and further detail of i faced.

Cloud Platform: `AWS EC2` (with Docker installed)
Storage: `AWS S3` with two bucket (c.f slidehub-images, slidehub-slides)
DB: `MySQL 5.7`
SQS: `AWS SQS`
Slidehub: `V1.6.2`

PDF: 
- Created and Converted by PowerPoint for Mac 16.12
- Created with Powerpoint provided template
- With Images
- With English & Traditional Chinese wordings
- With `SmartArt` (Powerpoint graph function) in one slide
- around 20 slides

I was first trying to upload .pptx file directly when i first use Slidehub but not work. When I use V1.5.x I encounter with specific slides. After update to new version all slides are not able to upload.

Hope this information helps to build this amazing repo.

**Comment by kevinlwhlwh** [comment]
Full Steps I init Slidehub:

Step 1 -- create mysql db
```bash
docker run --detach --name slidehub-db-mysql -p 3306:3306 -e MYSQL_ROOT_PASSWORD=xxxxxxx mysql:5.7
```

Step 2 -- Run Slidehub
```bash
docker run -d \
--restart=always \
--env RAILS_ENV=production \
--env RAILS_LOG_TO_STDOUT=1 \
--env OSS_REGION=ap-southeast-1 \
--env OSS_SQS_URL=https://sqs.ap-southeast-1.amazonaws.com/000000000/slidehub-sqs \
--env OSS_BUCKET_NAME=${SLIDEHUB_SLIDES_BUCKET} \
--env OSS_IMAGE_BUCKET_NAME=${SLIDEHUB_IMAGES_BUCKET} \
--env OSS_USE_S3_STATIC_HOSTING=0 \
--env OSS_AWS_SECRET_KEY=${AWS_SECRET_KEY} \
--env OSS_AWS_ACCESS_ID=${AWS_ACCESS_ID} \
--env OSS_SECRET_KEY_BASE=${GENERATED_SECREY_KEY} \
--env OSS_DB_NAME=slidehub \
--env OSS_DB_USERNAME=root \
--env OSS_DB_PASSWORD=${DB_PW} \
--env OSS_DB_URL=slidehub-db \
--env OSS_DB_PORT=3306 \
--env OSS_DB_USE_AZURE=false \
--env OSS_PRODUCTION_HOST=slidehub.example.com \
--env OSS_ROOT_URL=http://slidehub.example.com \
--link slidehub-db-mysql:slidehub-db \
--volume=$HOME/slidehub-tmp:/tmp \
-p 80:3000 --name slidehub ryuzee/slidehub:latest
```

Step 3 -- rails migrate db
```bash
docker exec ${SLIDEHUB_CONTAINER} bash -l -c 'bundle exec rake db:create RAILS_ENV=production'
docker exec ${SLIDEHUB_CONTAINER} bash -l -c 'bundle exec rake db:migrate RAILS_ENV=production'
docker exec ${SLIDEHUB_CONTAINER} bash -l -c 'bundle exec rake db:seed RAILS_ENV=production'
```

**Comment by ryuzee** [comment]
@kevinlwhlwh Your installation procedure looks good to me. Error messages indicate that the issue is coming from the pdf conversion to text.
From your description, pdf includes Chinese texts and I am going to dive into this point.
Please wait for a while to complete the investigation. Sorry for the bother.

**Comment by ryuzee** [comment]
Chinese texts can be handled with the app. I downloaded several slide decks that include Chinese characters and uploaded to the app. Conversion processes completed successfully...

**Comment by ryuzee** [comment]
@kevinlwhlwh I released V1.6.3 last night. From your comment, it seems you are using V1.6.2. Could you try to use the latest version?
If you are already using the latest version and the issue is not resolved, please provide me excerpt of logs again?

**Comment by kevinlwhlwh** [comment]
Thanks. Issue fixed with V1.6.3 trying to upload few pdfs and all worked. Issue can now closed
